nstack-kotlin-android/src/main/java/dk/nodes/nstack/kotlin/NStack.kt

```diff
@@ -718,7 +718,11 @@ object NStack {
      */
     private suspend fun <T> guardConnectivity(block: suspend () -> Result<T>): Result<T> {
         return if (connectionManager.isConnected) {
-            block()
+            try {
+                block()
+            } catch (e: Exception) {
+                Result.Error(Error.UnknownError)
```

Comment:
NetworkError no?

Reply:
We are checking network below, the exception could be due to other things - we could potentially include the exception for better tracing later on